packages/imperative/src/config/cmd/report-env/EnvQuery.ts

```diff
@@ -0,0 +1,624 @@
+                    if (IO.isDir(profileTypeDir)) {
+                        // does the next profile type directory have any profiles?
+                        fs.readdirSync(profileTypeDir).forEach((nextProfileNm) => {
+                            // exclude the meta files
+                            if (nextProfileNm.endsWith("_meta.yaml")) {
+                                return;
+                            }
+                            profilesOfCurrType += EnvQuery.indent + EnvQuery.indent +
+                                nextProfileNm.replace(".yaml", "") + os.EOL;
+                        });
+                    }
+
+                    // did we find any profiles?
+                    if (profilesOfCurrType.length > 0) {
+                        getResult.itemValMsg += EnvQuery.indent + nextProfileTypeNm +
+                        " profiles: " + os.EOL + profilesOfCurrType;
+
+                    }
+                });
+            }
+        }
+
+        // add indent to each line
+        getResult.itemValMsg  = EnvQuery.divider + "Zowe CLI configuration information:" +
+            os.EOL + os.EOL + EnvQuery.indent +
+            getResult.itemValMsg.replace(EnvQuery.allEolRegex, "$1" + EnvQuery.indent);
+
+        configProgress.statusMessage = "Complete";
+        configProgress.percentComplete = TaskProgress.ONE_HUNDRED_PERCENT;
+        configProgress.stageName = TaskStage.COMPLETE;
+        await EnvQuery.updateProgressBar(doesProgBarExist);
+        if (doesProgBarExist) {
+            getItemOpts.progressApi.endBar();
+        }
+    } // end getConfigInfo
+
+    // __________________________________________________________________________
+    /**
+     * For the specified itemId, get any known problems.
+     *
+     * @param itemId ID of the environmental item for which we want to detect problems.
+     * @param itemVal The value of the environmental item.
+     * @returns A string with a message about the problems. An empty string if no problems are detected.
+     */
+    private static getEnvItemProblems(itemId: ItemId, itemVal: string): string {
+        let probMsgs: string = "";
+        for (const nextProbTest of probTests) {
+            if (itemId === nextProbTest.itemId) {
+                if (EnvQuery.detectProbVal(itemVal, nextProbTest)) {
+                    if (probMsgs.length > 0) {
+                        probMsgs += os.EOL;
+                    }
+                    probMsgs += nextProbTest.probMsg;
+                }
+            }
+        }
+        return probMsgs;
+    }
+
+    // __________________________________________________________________________
+    /**
+     * Get information about the NPM configuration.
+     *
+     * @param getResult The itemVal and itemValMsg properties are filled
+     *                  by this function.
+     * @param getItemOpts options that affect our behavior.
+     */
+    private static async getNpmInfo(
```

Comment:
All of these npm commands are informational only, i.e. they don't modify the npmrc file

While it's not the case for these commands, please note that there are certain commands that rewrite npmrc thus removing comments.

I've confirmed that npm 9 does not write back to npmrc for any of these commands 😋